src/neo4j_genai/kg_construction/kg_writer.py

```diff
@@ -0,0 +1,138 @@
+#  Copyright (c) "Neo4j"
+#  Neo4j Sweden AB [https://neo4j.com]
+#  #
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#  #
+#      https://www.apache.org/licenses/LICENSE-2.0
+#  #
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+from __future__ import annotations
+
+from abc import abstractmethod
+from typing import Literal, Optional
+
+import neo4j
+from neo4j_genai.core.component import Component, DataModel
+from neo4j_genai.indexes import upsert_vector, upsert_vector_on_relationship
+from neo4j_genai.kg_construction.types import Neo4jGraph, Neo4jNode, Neo4jRelationship
+from neo4j_genai.neo4j_queries import UPSERT_NODE_QUERY, UPSERT_RELATIONSHIP_QUERY
+from pydantic import validate_call
+
+
+class KGWriterModel(DataModel):
+    status: Literal["SUCCESS", "FAILURE"]
+
+
+class KGWriter(Component):
+    """Abstract class used to write a knowledge graph to a data store."""
+
+    @abstractmethod
+    @validate_call
+    async def run(self, graph: Neo4jGraph) -> KGWriterModel:
+        """
+        Writes the graph to a data store.
+
+        Args:
+            graph (Neo4jGraph): The knowledge graph to write to the data store.
+        """
+        pass
+
+
+class Neo4jWriter(KGWriter):
+    """Writes a knowledge graph to a Neo4j database.
+
+    Args:
+        driver (neo4j.driver): The Neo4j driver to connect to the database.
+        neo4j_database (Optional[str]): The name of the Neo4j database to write to. Defaults to 'neo4j' if not provided.
+    """
+
+    def __init__(
+        self,
+        driver: neo4j.driver,
+        neo4j_database: Optional[str] = None,
```

Comment:
Should this default to "neo4j" as described in the docstring?

Reply:
Yeah it works the same was as `create_vector_index`, `create_fulltext_index`, and `drop_index_if_exists`, etc. from `src/neo4j_genai/indexes.py`. If `database` is set to `None` in the `execute_query` method of the Neo4j driver it defaults to using the `neo4j` database.